Formula/jenkins.rb

```diff
@@ -1,8 +1,8 @@
 class Jenkins < Formula
   desc "Extendable open source continuous integration server"
-  homepage "https://jenkins.io/"
-  url "http://mirrors.jenkins.io/war/2.313/jenkins.war"
-  sha256 "e41690114ed7bcb3efe719612edb4f6180c7ef5e76a3eaf66193edbc09a2f9d2"
+  homepage "https://www.jenkins.io/"
+  url "https://get.jenkins.io/war/2.314/jenkins.war"
```

Comment:
jenkins-lts already uses the correct url:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/jenkins-lts.rb#L4

Reply:
@timja thanks for the contribution, can you make the PR two commits?
- `jenkins-lts: update homepage`
- `jenkins 2.314`

So that the bot can do the automerge thing. Thanks!